ibm/service/cos/data_source_ibm_cos_bucket.go

```diff
@@ -707,31 +708,31 @@ func dataSourceIBMCosBucketRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("s3_endpoint_public", apiEndpoint)
 	d.Set("s3_endpoint_private", apiEndpointPrivate)
 	d.Set("s3_endpoint_direct", directApiEndpoint)
-
-	getBucketConfigOptions := &resourceconfigurationv1.GetBucketConfigOptions{
-		Bucket: &bucketName,
-	}
-
-	sess, err := meta.(conns.ClientSession).CosConfigV1API()
-	if err != nil {
-		return err
-	}
-
+	authenticator := new(core.IamAuthenticator)
```

Comment:
Why are we intializing authenticator here. It should be in config.go similar to CosConfigV1API()
Every time only APIkey is n't  used to work with provider. 
The provider can work with IAM tokens (in Schematics)